File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    # bh, bw = yh + 2, yw + 2
    # yellow + brown = bh * bw
    
    for bh in range(1, brown//2 + 1): # 길어봤자지 ~
        bw = (brown - 2*bh + 4)//2
        yh, yw = bh - 2, bw - 2
        if yellow == yh * yw and yellow + brown == bh * bw:
            return [bw, bh]

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int total = brown + yellow;

        for (int h = 3; h <= total / 3; h++) {
            int w = total / h;

            if (total % h == 0 && w >= h) {
                int y = (w - 2) * (h - 2);
                if (y == yellow) {
                    return new int[]{w, h};
                }
            }
        }

        return new int[]{0, 0};
    }
}